PayPalNativeCheckout/src/main/java/com/braintreepayments/api/PayPalNativeCheckoutClient.java

```diff
@@ -203,6 +203,8 @@ private void sendPayPalRequest(
                         braintreeClient.sendAnalyticsEvent("paypal-native.billing-agreement.succeeded");
                     }
                 });
+            } else {
+                listener.onPayPalFailure(new Exception(error));
```

Comment:
```suggestion
                listener.onPayPalFailure(error);
```
🧶 : could we forward the error here?

Reply:
we could but its marked nullable from the above method so I think its a bit safer to create a new error here instead of just forwarding it to the listener and then making the listeners error nullable instead of non null

Reply:
added test `requestNativeCheckout_returnsErrorFromFailedResponse`